File: backend-java/persona.java
public class Persona {
    private String nombre;
    private int edad;
    private String email;

    public Persona(String nombre, int edad, String email) {
        if (nombre == null || nombre.trim().isEmpty()) {
            throw new IllegalArgumentException("El nombre no debe estar vacío.");
        }
        if (edad < 0) {
            throw new IllegalArgumentException("La edad debe ser mayor o igual a 0.");
        }
        if (email == null || !email.contains("@")) {
            throw new IllegalArgumentException("El email debe contener '@'.");
        }

        this.nombre = nombre;
        this.edad = edad;
        this.email = email;
    }

    public boolean esMayorDeEdad() {
        return edad >= 18;
    }

    public void mostrarInfo() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Edad: " + edad);
        System.out.println("Email: " + email);
    }
}

File: backend-python/persona.py
class Persona:
    def __init__(self, nombre, edad, email):
        if edad <= 0:
            raise ValueError("La edad debe ser positiva.")
        if "@" not in email:
            raise ValueError("El email debe contener '@'.")
        self.nombre = nombre
        self.edad = edad
        self.email = email

    def es_mayor_de_edad(self):
        return self.edad >= 18

    def mostrar_info(self):
        print(f"Nombre: {self.nombre}")
        print(f"Edad: {self.edad}")
        print(f"Email: {self.email}")


if __name__ == "__main__":
    persona = Persona("Ilan", 21, "ilan@example.com")
    persona.mostrar_info()
    print("¿Mayor de edad?", persona.es_mayor_de_edad())
